src/java.base/share/classes/java/util/stream/Gatherer.java

```diff
@@ -0,0 +1,593 @@
+     * provided {@link Downstream}.
+     *
+     * @return a function which integrates provided elements, potentially using
+     *         the provided state, optionally producing output to the provided
+     *         Downstream
+     */
+    Integrator<A, T, R> integrator();
+
+    /**
+     * A function which accepts two intermediate states and combines them into
+     * one.
+     *
+     * @implSpec The implementation in this interface returns
+     *           {@link #defaultCombiner()}.
+     *
+     * @return a function which accepts two intermediate states and combines
+     *         them into one
+     */
+    default BinaryOperator<A> combiner() {
+        return defaultCombiner();
+    }
+
+    /**
+     * A function which accepts the final intermediate state
+     * and a {@link Downstream} object, allowing to perform a final action at
+     * the end of input elements.
+     *
+     * @implSpec The implementation in this interface returns
+     *           {@link #defaultFinisher()}.
+     *
+     * @return a function which transforms the intermediate result to the final
+     *         result(s) which are then passed on to the provided Downstream
+     */
+    default BiConsumer<A, Downstream<? super R>> finisher() {
+        return defaultFinisher();
+    }
+
+    /**
+     * Returns a composed Gatherer which connects the output of this Gatherer
+     * to the input of that Gatherer.
+     *
+     * @implSpec The implementation in this interface returns a new Gatherer
+     *           which is semantically equivalent to the combination of
+     *           {@code this} and {@code that} gatherer.
+     *
+     * @param that the other gatherer
+     * @param <RR> The type of output of that Gatherer
+     * @throws NullPointerException if the argument is {@code null}
+     * @return returns a composed Gatherer which connects the output of this
+     *         Gatherer as input that Gatherer
+     */
+    default <RR> Gatherer<T, ?, RR> andThen(Gatherer<? super R, ?, ? extends RR> that) {
+        Objects.requireNonNull(that);
+        return Gatherers.Composite.of(this, that);
+    }
+
+    /**
+     * Returns an initializer which is the default initializer of a Gatherer.
+     * The returned initializer identifies that the owner Gatherer is stateless.
+     *
+     * @implSpec This method always returns the same instance.
+     *
+     * @see Gatherer#initializer()
+     * @return the instance of the default initializer
+     * @param <A> the type of the state of the returned initializer
+     */
+    static <A> Supplier<A> defaultInitializer() {
+        return Gatherers.Value.DEFAULT.initializer();
+    }
+
+    /**
+     * Returns a combiner which is the default combiner of a Gatherer.
+     * The returned combiner identifies that the owning Gatherer must only
+     * be evaluated sequentially.
+     *
+     * @implSpec This method always returns the same instance.
+     *
+     * @see Gatherer#finisher()
```

Comment:
Today I noticed that this is the wrong link. It should be `Gatherer#combiner()`, not `Gatherer#finisher()`.

Reply:
@robtimus Thanks! I'll address that 👍